backend/workflow_manager/endpoint/database_utils.py

```diff
@@ -284,22 +299,76 @@ def execute_write_query(
         - Snowflake does not support INSERT INTO ... VALUES ...
           syntax when VARIANT columns are present (JSON).
           So we need to use INSERT INTO ... SELECT ... syntax
+        - sql values can contain data with single quote. It needs to
         """
-        sql = (
-            f"INSERT INTO {table_name} ({','.join(sql_keys)}) "
-            f"SELECT {','.join(sql_values)}"
-        )
-        logger.debug(f"insertng into table with: {sql} query")
+        sql = DBConnectorQueryHelper.build_sql_insert_query(table_name, sql_keys)
+        logger.debug(f"inserting into table {table_name} with: {sql} query")
+
+        sql_values = DBConnectorQueryHelper.prepare_sql_values(cls_name, sql_values)
+        logger.debug(f"sql_values: {sql_values}")
+
         try:
-            if hasattr(engine, "cursor"):
-                with engine.cursor() as cursor:
-                    cursor.execute(sql)
-                engine.commit()
+            if cls_name == DBConnectionClass.BIGQUERY:
```

Comment:
@kirtimanmishrazipstack a cleaner way to implement this might be to make use of a base connector abstraction that declares an execute method. All concrete classes (bigquery, mysql .etc.) can implement this abstraction and handle its corresponding exception.
With the current changes we'll have an ever increasing list of such exceptions / methods with `if` and `elif`.

To bring more analogy - try referring to the test DB connector call and see how errors are handled

Reply:
@chandrasekharan-zipstack 

It is not possible to move execute_query to bases class because , some class might raise same exception while other might raise different exception.
For eg: postgres and redshift will raise same psycopg error, but snowflake will raise different exception. So there will be duplcation in postgres and redshoft. One way to solve is to have a common pyscopg connector class which  postgres and redshift will inherit. But I dont want to create a new class in db connector. So let database_utils be same.

Reply:
@kirtimanmishrazipstack its okay if separate exceptions are raised from those classes - we just catch it within the concrete class implementation and re-throw the exception as a `ConnectorError`. In the backend we just catch `ConnectorError` and populate the appropriate API message